src/transport/SecureSessionMgr.h

```diff
@@ -53,7 +53,7 @@ namespace chip {
  *  EncryptedPacketBufferHandle is a kind of PacketBufferHandle class and used to hold a packet buffer
  *  object whose payload has already been encrypted.
  */
-class EncryptedPacketBufferHandle final : private System::PacketBufferHandle
+class EncryptedPacketBufferHandle final : public System::PacketBufferHandle
```

Comment:
This was private so people couldn't accidentally start poking around at the buffer of an EncryptedPacketBufferHandle and modifying it.  Except SecureSessionMgr needs to do it, for the moment, hence it was a friend.

I would prefer that we keep this private, keep `SecureSessionMgr` as a friend, add `using System::PacketBufferHandle::IsNull;` so the null-test in reliable message manager can work....

Reply:
OK, talked to @andy31415 .  Let's land this as-is and I'll do a followup to put back in the safeguards.